x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDeploymentStatsAction.java

```diff
@@ -0,0 +1,289 @@
+
+    private GetDeploymentStatsAction() {
+        super(NAME, GetDeploymentStatsAction.Response::new);
+    }
+
+    public static class Request extends BaseTasksRequest<GetDeploymentStatsAction.Request> {
+
+        public static final String ALLOW_NO_MATCH = "allow_no_match";
+
+        private final String deploymentId;
+        private boolean allowNoMatch = true;
+        // used internally this should not be set by the REST request
+        private List<String> expandedIds;
+
+        public Request(String deploymentId) {
+            this.deploymentId = ExceptionsHelper.requireNonNull(deploymentId, "deployment_id");
+            this.expandedIds = Collections.singletonList(deploymentId);
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.deploymentId = in.readString();
+            this.allowNoMatch = in.readBoolean();
+            this.expandedIds = in.readStringList();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(deploymentId);
+            out.writeBoolean(allowNoMatch);
+            out.writeStringCollection(expandedIds);
+        }
+
+        public String getDeploymentId() {
+            return deploymentId;
+        }
+
+        public List<String> getExpandedIds() {
+            return expandedIds;
+        }
+
+        public void setExpandedIds(List<String> expandedIds) {
+            this.expandedIds = expandedIds;
+        }
+
+        public void setAllowNoMatch(boolean allowNoMatch) {
+            this.allowNoMatch = allowNoMatch;
+        }
+
+        public boolean isAllowNoMatch() {
+            return allowNoMatch;
+        }
+
+        @Override
+        public boolean match(Task task) {
+            return expandedIds.stream().anyMatch(taskId -> StartTrainedModelDeploymentAction.TaskMatcher.match(task, taskId));
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(deploymentId, request.deploymentId) &&
+                this.allowNoMatch == request.allowNoMatch &&
+                Objects.equals(expandedIds, request.expandedIds);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(deploymentId, allowNoMatch, expandedIds);
+        }
+    }
+
+
+    public static class Response extends BaseTasksResponse implements ToXContentObject {
+
+        public static final ParseField DEPLOYMENT_STATS = new ParseField("deployment_stats");
+
+        public static class AllocationStats implements ToXContentObject, Writeable {
+
+            public static class NodeStats implements ToXContentObject, Writeable {
```

Comment:
This should have `RoutingState` and a `reason` object here. Its conceivable that a node is failed, stopping, etc. and provides these stats, but the current state is not accepting inference calls.